### PR TITLE
Remove include/CMakeLists.txt

### PR DESCRIPTION
**Pull request by jpr42**

### CODE REVIEW

File: include/CMakeLists.txt [hunk 1]
```diff
@@ -1,13 +0,0 @@
-# Add include folders to the library and targets that consume it
-# the SYSTEM keyword suppresses warnings for users of the library
-#
-# By adding this directory as an include directory the user gets a
-# namespace effect.
-#
-# IE:
-#   #include <gsl/gsl>
```
Comment (jpr42) [inline include/CMakeLists.txt:8]: CI testing ensures this namespace effect is tested.

File: include/CMakeLists.txt [hunk 2]
```diff
@@ -1,13 +0,0 @@
...
-if(PROJECT_IS_TOP_LEVEL)
-    target_include_directories(GSL INTERFACE $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>)
-else()
-    target_include_directories(GSL SYSTEM INTERFACE $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>)
```
Comment (jpr42) [inline include/CMakeLists.txt:12]: Using the SYSTEM keyword is not necessary anymore.

It was never needed for find_package support.

It only benefited `add_subdirectory` but now with CMake 3.25 it's not needed anymore.

```
New in version 3.25: If the SYSTEM argument is provided, the [SYSTEM](https://cmake.org/cmake/help/latest/prop_dir/SYSTEM.html#prop_dir:SYSTEM) directory property of the subdirectory will be set to true. This property is used to initialize the [SYSTEM](https://cmake.org/cmake/help/latest/prop_tgt/SYSTEM.html#prop_tgt:SYSTEM) property of each non-imported target created in that subdirectory.
```

Comment (jpr42) [reply include/CMakeLists.txt:12]: To be clear CMake 3.25 also lets you choose whether you even need the SYSTEM keyword anyway.

Comment (jpr42) [reply include/CMakeLists.txt:12]: Also using the SYSTEM keyword makes local GSL development more difficult since it ignores warnings from header files.

Comment (dmitrykobets-msft) [reply include/CMakeLists.txt:12]: the minimum-supported cmake version is 3.14 though, no?

Comment (jpr42) [reply include/CMakeLists.txt:12]: > the minimum-supported cmake version is 3.14 though, no?

Not sure where my head was with this PR... apologies. See new PR.

**Pull request closed without merging**